### PR TITLE
Use HintPath instead of direct reference for clearer reference display in VS

### PR DESCRIPTION
**Pull request by thinkbeforecoding**
Hi pcm !
Here is a proposal to use HintPath instead on full path in references. This helps having cleaner display in VS references.
It also moves the selection of nuget/project ref to a <Choose><When> <Otherwhise></Choose> structure to help ncrunch analysis.

Cheers,

Jeremie

### CODE REVIEW

**Comment by pchalamet** [comment]
:white_check_mark: [Build full-build 4.2.31 completed](https://ci.appveyor.com/project/full-build/full-build-shu78/build/4.2.31) (commit https://github.com/full-build/full-build/commit/de1a6e8d36 by @)

**Comment by pchalamet** [comment]
Hi, thanks. Can't really see what this will change in VS UI (it's not MSBuild and dependency analysis is too light). Maybe a screenshots before and after could help seeing how experience will change?
Looks good btw. 

@sergii-s can you review and check everything is fine ?

**Comment by thinkbeforecoding** [comment]
The Reference is now like:
+ References
+- MyCompany.MyLib

instead of:
+ References
+- c:/fullbuild//.fullbuild/bin/MyCompany.MyLib.dll

**Comment by pchalamet** [comment]
👍

File: src/FullBuild/Generators/MsBuild.fs [hunk 1]
```diff
@@ -67,16 +66,22 @@ let private generateProjectTarget (project : Project) =
...
-                    XAttribute (NsNone + "Include", binFile),
-                    XAttribute (NsNone + "Condition", binCondition),
-                    XElement (NsMsBuild + "Private", "true"))),
+            XElement(NsMsBuild + "Choose",
+                XElement(NsMsBuild + "When", 
+                    XAttribute(NsNone + "Condition", srcCondition),
+                    XElement (NsMsBuild + "ItemGroup",
+                        XElement(NsMsBuild + "ProjectReference",
+                            XAttribute (NsNone + "Include", projectFile),
+                            //XAttribute (NsNone + "Condition", srcCondition),
```
Comment (pchalamet) [inline src/FullBuild/Generators/MsBuild.fs]: remove useless comment

File: src/FullBuild/Generators/MsBuild.fs [hunk 2]
```diff
@@ -67,16 +66,22 @@ let private generateProjectTarget (project : Project) =
...
+                            XElement (NsMsBuild + "Project", sprintf "{%s}" project.UniqueProjectId),
+                            XElement (NsMsBuild + "Name", project.Output.Name)))),
+                XElement(NsMsBuild + "Otherwise",    
+                    XElement(NsMsBuild + "ItemGroup",
+                        XElement (NsMsBuild + refType,
+                            XAttribute (NsNone + "Include", project.Output.Name),
+                            //XAttribute (NsNone + "Condition", binCondition),
```
Comment (pchalamet) [inline src/FullBuild/Generators/MsBuild.fs]: remove this too

**Comment by pchalamet** [comment]
+ add notes in CHANGES.md (create a 4.3 section please)

**Comment by thinkbeforecoding** [comment]
Done 👍

**Comment by pchalamet** [comment]
:white_check_mark: [Build full-build 4.2.32 completed](https://ci.appveyor.com/project/full-build/full-build-shu78/build/4.2.32) (commit https://github.com/full-build/full-build/commit/6cb1a22f42 by @)

**Comment by pchalamet** [comment]
remove commented code too ;-)

**Comment by pchalamet** [comment]
@sergii-s : can you then review and validate merge please ?

**Comment by pchalamet** [comment]
ok for me - don't know about you @sergii-s

**Comment by pchalamet** [comment]
:white_check_mark: [Build full-build 4.2.33 completed](https://ci.appveyor.com/project/full-build/full-build-shu78/build/4.2.33) (commit https://github.com/full-build/full-build/commit/dd7305aa02 by @)

**Comment by sergii-s** [comment]
👍

**Pull request merged**